idaes/core/util/tables.py

```diff
@@ -266,22 +267,47 @@ def stream_table_dataframe_to_string(stream_table, **kwargs):
         na_rep=na_rep, justify=justify, float_format=float_format, **kwargs
     )
 
+def _get_state_from_port(port,time_point):
+    """
+    Attempt to get a state-block-like object connected to a Port. If different
+    variables on the port appear to be connected to different state blocks,
+    raise an exception. This process is complicated by the fact that the object
+    may be indexed by space as well as time. Therefore, we assume that, if it
+    is, the time index comes first.
+    """
+    vlist = [v for v in port.iter_vars()]
+    states = [v.parent_block().parent_component() for v in vlist]
 
-def _get_state_from_port(port, time_point):
-    # Check port for _state_block attribute
+    if len(vlist) == 0:
+        raise AttributeError(
```

Comment:
I would make this a ValueError - it is occurring due to a lack of "values" in the Port. This will also help distinguish the different types of failures that can occur, so that users of this method can decide how to handle the different cases if needed.